packages/rrweb-snapshot/test/css.test.ts

```diff
@@ -119,6 +119,64 @@ describe('css parser', () => {
     expect(out3).toEqual('[data-aa\\:other] { color: red; }');
   });
 
+  it.each([
+    ['.foo,.bar {}', ['.foo', '.bar']],
+    ['.bar:has(:disabled) {}', ['.bar:has(:disabled)']],
+    [
+      '.bar:has(input:is(:disabled),button:has(:disabled)) {}',
+      ['.bar:has(input:is(:disabled),button:has(:disabled))'],
+    ],
+    [
+      '.bar:has(input:is(:disabled),button:has(:disabled,.baz)) {}',
+      ['.bar:has(input:is(:disabled),button:has(:disabled,.baz))'],
+    ],
```

Comment:
I'd also do a simpe-ish test where the first chunk has 1 unbalanced opening brace, the 2nd chunk is balanced also with 1, and the 3rd is the closing for the initial chunk..

```suggestion
    [
      '.bar:has(div, input:is(:disabled), button) {}',
      ['.bar:has(div, input:is(:disabled), button)'],
    ],
    [
      '.bar:has(div, input:is(:disabled),button:has(:disabled,.baz)) {}',
      ['.bar:has(div, input:is(:disabled),button:has(:disabled,.baz))'],
    ],```